cmake/modules/extensions.cmake

```diff
@@ -2481,76 +2481,76 @@ Please provide one of following: APPLICATION_ROOT, CONF_FILES")
     set(multi_args CONF_FILES NAMES)
   endif()
 
-  cmake_parse_arguments(FILE "${options}" "${single_args}" "${multi_args}" ${ARGN})
-  if(FILE_UNPARSED_ARGUMENTS)
-      message(FATAL_ERROR "zephyr_file(${ARGV0} <val> ...) given unknown arguments: ${FILE_UNPARSED_ARGUMENTS}")
+  cmake_parse_arguments(ZFILE "${options}" "${single_args}" "${multi_args}" ${ARGN})
```

Comment:
Instead of creating a new commit with combination of multiple commits, then I believe the proper way is to actually do a cherry-pick of needed commits from `zephyr/main`.

That will both make it easier to identify identical changes / commits between branches, as well as help identifying cases where some commits might have been forgotten.

In this case it appears to me that the changes in 6ebb408921c535c45ca4c31b9ded672a0dae456c lacks the fixes from 3cff550180fe83849d6029c93865e088df361617, and hence while trying to fix #71203 you introduce a new error.

Please cherry-pick relevant commits instead, which seems to be the following:
50f0454fbd446b5bb0653b25cedccd022aa18006
3cff550180fe83849d6029c93865e088df361617
030130573a035f50f0cc931763e4767dd3e8ab3b (from PR #71280)

Reply:
It is not cherry-picked because it does not apply, the changes in those linked commits use MERGE functions which are not in 3.6

Reply:
I know it's not cherry-picked but you can still cherry-pick and during conflict resolving you'll get the chance to adjust the commit wrt. the MERGE functionality and the board qualifiers.

And the `-x` makes it even easier to see the origin of the fix.

End result in code should of course be identical to re-implementing the changes, but making it much easier to trace which fixes from main that has been applied in the branch.